### ISSUE TITLE
Inappropriate leaves being removed, presumably by lumberjacks.

### ISSUE THREAD

**Issue by Shamshalabim** [TangoTek/TekTopia-Community#524]
**Describe the bug**
I've noticed many leaf blocks in the floor disappearing. I've yet to actually see it happen. Assuming that it is something to do with the lumberjacks!

**To Reproduce**
1. Plant many many leaf blocks flat in the floor where villagers walk..
2. Notice that some of the leaf blocks disappear at random.

**Expected behavior**
I'd expect leaf blocks that are not part of a tree to not be left alone. 

**Screenshots**
Not really applicable. Just leaf blocks missing from the ground leaving holes. The idea was to have lighting under them..

** Server/Location **
Server 1
-3714, 71, 2018

**Additional context**
I might suggest that a solution might be to prevent leaf removal if the leaf is on top of a light source blocks... That said, I don't know for sure that lumberjacks are the cause of this removal..

**Comment by Sobertron** [comment]
I think it has to do with the lumberjacks removal of leafs being always on. If you look at the image below the leaves on flat ground are not getting destroyed but the ones near the stairs are. I believe with their current ai they are unintentionally destroying these leaves as they walk

![2019-06-24_17 22 11](https://user-images.githubusercontent.com/49371154/60035515-ead2d780-96a4-11e9-83bc-049687e4feed.png)

**Comment by Shamshalabim** [comment]
Yes, I'd agree with Sober's assumption, but interestingly it doesn't happen every time. I've seen them walk up and down the stairs many times and not damage the leaves... I'd be more interested to know how the leaf removal works a bit better.. Could this functionality maybe be folded into the scan for trees somehow?? Like it gets turned on when they 'locate' target tree, and off when they successfully cut it down??

**Comment by Shamshalabim** [comment]
I want to do a bunch of groundwork atm in our village to really bring it all together, but if it's going to be getting deleted all the time, I'll wait.

**Comment by LadyGwynn** [comment]
I'm not at all familiar with coding, or even with what NBT data is or how items differ from other items of a similar type in Minecraft... but it seems to me that there IS something different about leaves placed by a player vs naturally spawned leaves, because the leaves placed by a player won't despawn when there is no wooden block adjacent to them. Would it be possible to limit a lumberjack's ability to break leaves to only those leaves that are naturally spawned? This would make it easier to decorate our villages with bushes and the like in addition to the hidden lighting that was mentioned in the above posts.

**Comment by TheGreatNaj** [comment]
That might make sense of whats been going on with my trees too actually. I've been unsure about posting this because it might be unrelated to the mod but leaves have been disappearing from other trees not chopped down 
![2019-06-25_04 45 12](https://user-images.githubusercontent.com/25763105/60100323-7f7e1980-9727-11e9-91c2-d9b4bddd4390.png)
